### PR TITLE
Convert project to netstandard/net6.

### PR DESCRIPTION
**Pull request by Rivsoft**
Migrate to SDK style project structure.
Fix tests, assembly reflection loading among a few other things.

### CODE REVIEW

**Comment by PMudra** [comment]
Thanks for your contribution @Rivsoft. Are you somehow connected to @Nico-1987 who created the issue #26?

As I mentioned in the issue, I am not able to review these changes (not enough time and knowledge).

@swiebertje do you have the capacity to review these changes?

**Comment by swiebertje** [comment]
> Thanks for your contribution @Rivsoft. Are you somehow connected to @Nico-1987 who created the issue #26?
> 
> As I mentioned in the issue, I am not able to review these changes (not enough time and knowledge).
> 
> @swiebertje do you have the capacity to review these changes?

Hi @PMudra, that's fine, I will have a look at it. 

@Rivsoft, @Nico-1987 and me actually work at the same company 😃

**Comment by Rivsoft** [comment]
Hi all:

Thanks for the reply. Regarding the PR I think the only issue I noticed atm
is the appveyor build failing trying to restore the fake app (using the
build.bat script). I never worked with appveyor before but I assume this
should be a simple fix to implement. If someone can take a look at it that
would be great, otherwise let me know and I will see if I can take a stab
at it.

Thanks in advance

**Comment by swiebertje** [comment]
> Hi all: Thanks for the reply. Regarding the PR I think the only issue I noticed atm is the appveyor build failing trying to restore the fake app (using the build.bat script). I never worked with appveyor before but I assume this should be a simple fix to implement. If someone can take a look at it that would be great, otherwise let me know and I will see if I can take a stab at it. Thanks in advance
> […](#)

I updated the AppVeyor build environment from VS2015 to VS2022. Build is passing now.

File: ApiCheck/Loader/AssemblyLoader.cs [hunk 1]
```diff
@@ -15,21 +15,21 @@ public void Dispose()
...
       string path = Path.Combine(Path.GetDirectoryName(args.RequestingAssembly.Location), assemblyName.Name + ".dll");
       if (File.Exists(path))
       {
-        return Assembly.ReflectionOnlyLoadFrom(path);
+        return Assembly.LoadFile(path);
       }
       return Assembly.ReflectionOnlyLoad(AppDomain.CurrentDomain.ApplyPolicy(args.Name));
     }
 
     public Assembly ReflectionOnlyLoad(string path)
```
Comment (swiebertje) [inline ApiCheck/Loader/AssemblyLoader.cs:30]: Method name does not match the implementation anymore.

File: ApiCheck/Loader/AssemblyLoader.cs [hunk 2]
```diff
@@ -15,21 +15,21 @@ public void Dispose()
...
     {
-      return Assembly.ReflectionOnlyLoadFrom(path);
+      return Assembly.LoadFile(Path.GetFullPath(path));
```
Comment (swiebertje) [inline ApiCheck/Loader/AssemblyLoader.cs:32]: By using `Assembly.LoadFile` the assembly will be loaded in the AppDomain for code execution which has stricter rules than `ReflectionOnlyLoadFrom`. For example dependencies will also need to be loaded. However, we only need to load the assembly for inspecting it's contents. As an alternative to reflection we can have a look at [MetadataLoadContext](https://learn.microsoft.com/en-us/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext)

File: ApiCheck.NUnit/ApiTest.cs [hunk 3]
```diff
@@ -47,11 +41,11 @@ private IEnumerable<ApiTestData> GetResults()
         {
           ApiTestAttribute apiTestAttribute = (ApiTestAttribute)customAttribute;
 
-          Stream comparerConfigurationStream = GetReadStream(apiTestAttribute.ComparerConfigurationPath);
+          Stream comparerConfigurationStream = GetReadStream(Path.GetFullPath(apiTestAttribute.ComparerConfigurationPath, Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)));
```
Comment (swiebertje) [inline ApiCheck.NUnit/ApiTest.cs:44]: Why does the relative path not work anymore? And this conversion from relative to absolute path is duplicated in several places. Can be moved to some utility class.

File: ApiCheck.NUnit/ApiTest.cs [hunk 4]
```diff
@@ -8,13 +8,15 @@
...
 using System.Linq;
 using ApiCheck.Configuration;
+using NUnit.Framework.Interfaces;
+using System.Runtime.CompilerServices;
+using System.Reflection;
 
 namespace ApiCheck.NUnit
 {
-  public abstract class ApiTest
+  public static class ApiTest
```
[the remark this answers is not in the document]
Comment (Rivsoft) [reply ApiCheck.NUnit/ApiTest.cs:17]: I agree. I couldn't find a better solution that could preserve current API to be honest. 

Feel free to suggest one or maybe we just need to rethink the whole interface. To be honest I am not even entirely sure what's the purpose of that project, seems to be the base for the unit tests only ? If so should we just migrate it to the Tests projects ? Could definitely use some guidance here.

File: ApiCheck/Loader/AssemblyLoader.cs [hunk 5]
```diff
@@ -15,21 +15,21 @@ public void Dispose()
     {
       AppDomain.CurrentDomain.ReflectionOnlyAssemblyResolve -= CurrentDomainOnReflectionOnlyAssemblyResolve;
```
[the remark this answers is not in the document]
Comment (Rivsoft) [reply ApiCheck/Loader/AssemblyLoader.cs:16]: I agree, I am not sure we might to migrate the whole loader to use MetadataContext tbh but for the most part I was focused on making it work than refactor everything.

File: ApiCheck/Loader/AssemblyLoader.cs (hunk 1 above)
Comment (Rivsoft) [reply ApiCheck/Loader/AssemblyLoader.cs:30]: Same as above.

File: ApiCheck.NUnit/ApiTest.cs (hunk 3 above)
Comment (Rivsoft) [reply ApiCheck.NUnit/ApiTest.cs:44]: If I remember correctly the execution context was different than the original code so the current directory path is different now. This was breaking the tests.

File: ApiCheck.NUnit/ApiTest.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (PMudra) [reply ApiCheck.NUnit/ApiTest.cs:17]: > To be honest I am not even entirely sure what's the purpose of that project, seems to be the base for the unit tests only ? If so should we just migrate it to the Tests projects ? Could definitely use some guidance here.

The purpose of the `ApiCheck.NUnit` project is to offer a simple integration for unit test framworks (NUnit in that case). The idea is that an author of a library is able to easily add the api check to the existing unit tests of the library. For each check in the api check, a unit test will be generated. Failing tests will indicate changes in the api. Many library projects already have CI set up, so adding the NUnit integration was designed as an easy integration without the need of another CI step.

**Comment by Rivsoft** [comment]
Ok, makes sense. This breaking change will probably mean then we might need
a better way to integrate the library with NUnit.

File: ApiCheck.NUnit/ApiTest.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (swiebertje) [reply ApiCheck.NUnit/ApiTest.cs]: I restored the `ApiTest` base class functionality by moving the testcase builder code to a new custom NUnit attribute `ApiTestCaseSourceAttribute`. This attribute replaces `TestCaseSource` on the `ApiTest.ApiElementTest` method.

File: ApiCheck.NUnit/ApiTest.cs (hunk 3 above)
Comment (swiebertje) [reply ApiCheck.NUnit/ApiTest.cs]: I removed unneeded path conversions.

**Comment by swiebertje** [comment]
I published the NuGet package containing these changes: 2.0.107

**Pull request merged**